### ISSUE TITLE
Make throttled function cancellable

### ISSUE THREAD

**Issue by ofagbemi** [pelotoncycle/frame-throttle#3]
I want to prevent a throttled function's callback from being executed.

Lodash's `throttle` provides this functionality —
```javascript
import throttle from 'lodash/throttle'


const throttled = throttle(() => console.log('hi'), 100)

window.addEventListener('popstate', () => {
  throttled.cancel()
  console.log('cancelled throttled function');
});
```

Let me know if there's anything I can help with / clarify! And thanks for the great library!

**Comment by devrelm** [comment]
@ofagbemi 

Oh wow, I am so sorry for the late reply. I completely missed the notification about this issue.

Great idea, though! I'll see what I can whip up.

**Comment by devrelm** [comment]
One bit of clarification, though: are you looking for something which will cancel the next invocation or all following invocations?

**Comment by ntucker** [comment]
I think in the example those would both be fine, since it is navigating away this is just to clean up anything  that was queued since the component will be unmounted.

**Comment by devrelm** [comment]
@ofagbemi @ntucker Sorry again for taking so long to get this out. I've just published v3.0.0 with the new method. Check out [the readme](/pelotoncycle/frame-throttle/#readme) for how to use it, though it's basically as you described.

Cheers, and thanks!
